src/databricks/labs/ucx/hive_metastore/table_migrate.py

```
@@ -182,3 +184,144 @@ def print_revert_report(self, *, delete_managed: bool) -> bool | None:
+        logger.info(f"moved {len(list(table_tasks))} tables to the new schema {to_schema}.")
+        Threads.strict("creating views", view_tasks)
+        logger.info(f"moved {len(list(view_tasks))} views to the new schema {to_schema}.")
+
+    def _move_table(
+        self,
+        from_catalog: str,
+        from_schema: str,
+        from_table: str,
+        to_catalog: str,
+        to_schema: str,
+        del_table: bool,  # noqa: FBT001
+    ) -> bool:
+        from_table_name = f"{from_catalog}.{from_schema}.{from_table}"
+        to_table_name = f"{to_catalog}.{to_schema}.{from_table}"
+        try:
+            create_sql = str(next(self._backend.fetch(f"SHOW CREATE TABLE {from_table_name}"))[0])
+            create_table_sql = create_sql.replace(f"CREATE TABLE {from_table_name}", f"CREATE TABLE {to_table_name}")
+            logger.debug(f"Creating table {to_table_name}.")
+            self._backend.execute(create_table_sql)
+            grants = self._ws.grants.get(SecurableType.TABLE, from_table_name)
+            if grants.privilege_assignments is None:
+                return True
+            grants_changes = [
+                PermissionsChange(pair.privileges, pair.principal) for pair in grants.privilege_assignments
+            ]
+            self._ws.grants.update(SecurableType.TABLE, to_table_name, changes=grants_changes)
+            if del_table:
+                logger.info(f"dropping source table {from_table_name}")
+                drop_sql = f"DROP TABLE {from_table_name}"
+                self._backend.execute(drop_sql)
+            return True
+
+        except NotFound as err:
+            if "[TABLE_OR_VIEW_NOT_FOUND]" in str(err):
+                logger.error(f"Could not find table {from_table_name}. Table not found.")
+            else:
+                logger.error(err)
+        except PermissionDenied:
+            logger.error(f"error applying permissions for table {to_table_name}")
```

```suggestion
```
you shouldn't swallow permission errors.